metrics/stackdriver_exporter.go

```diff
@@ -115,3 +160,44 @@ func getMetricTypeFunc(metricTypePrefix, customMetricTypePrefix string) func(vie
 		return path.Join(customMetricTypePrefix, view.Measure.Name())
 	}
 }
+
+// getStackdriverSecret returns the Kubernetes Secret specified in the given config.
+func getStackdriverSecret(sdconfig *stackdriverClientConfig) (*corev1.Secret, error) {
+	err := ensureKubeclient()
+	if err != nil {
+		return nil, err
+	}
+
+	ns := secretNamespaceDefault
```

Comment:
Slight preference for writing this in the opposite way:

```golang
ns = sdconfig.GCPSecretNamespace
if ns == "" {
  ns = secretNamespaceDefault
}
```

Mostly because it focuses on the fact that you want `ns` to have a non-empty-string value, rather than that you care about `sdconfig` specifically.

Reply:
Pending